File: ui_webdriver/tests/pages/checkout.ts
import { storeValues } from '../support/constant.js';

async function searchArrayInList(nameKey: string, myArray: any) {
  return myArray.filter((i: any) =>  i.name === nameKey )[0]
}

// ASSERTSIONS //
async function validateItemsInCart(storedValues: any) {
  const nameResult = await browser
    .$(
      `//div[contains(text(), "${storedValues.title}")]//ancestor::div[@class="cart_item"]//div[@class="inventory_item_name"]`,
    )
    .getText();
  await expect(nameResult).toEqual(storedValues.title);

  const priceResult = await browser
    .$(
      `//div[contains(text(), "${storedValues.title}")]//ancestor::div[@class="cart_item"]//div[@class="inventory_item_price"]`,
    )
    .getText();
  await expect(priceResult).toEqual(storedValues.price);
}

export async function ValidateTotalCart(itemTotal: string) {
  const results = await browser.$$('[class="cart_item"]').length;
  await expect(results).toEqual(Number(itemTotal));
}

export async function ValidateItemInCart(itemValue: string) {
  const storedValues = await searchArrayInList(itemValue, storeValues);
  await validateItemsInCart(storedValues);
}

export async function ValidatesAllInCart() {
  for (let i = 0; i < storeValues.length; i++) {
    await validateItemsInCart(storeValues[i]);
  }
}

File: ui_selenium/features/pages/checkout.py
from selenium.webdriver.common.by import By
from features.support.constant import store_values
from features.support.helpers import validate

def search_array_in_list(name_key: str, my_array: any):
	return [x for x in my_array if x['name'] == name_key][0]

# // ASSERTSIONS //
def validate_items_in_cart(context, stored_values: any):
	name_result = context.driver.find_element(By.XPATH, f'//div[contains(text(), "{stored_values["title"]}")]//ancestor::div[@class="cart_item"]//div[@class="inventory_item_name"]')
	validate(stored_values["title"], name_result.text)

	price_result = context.driver.find_element(By.XPATH, f'//div[contains(text(), "{stored_values["title"]}")]//ancestor::div[@class="cart_item"]//div[@class="inventory_item_price"]')
	validate(stored_values["price"], price_result.text)

def validate_total_cart(context, item_total: str):
	results = context.driver.find_elements(By.CSS_SELECTOR, '[class="cart_item"]')
	validate(int(item_total), len(results))

def validate_item_in_cart(context, item_value: str):
	stored_values = search_array_in_list(item_value, store_values)
	validate_items_in_cart(context, stored_values)

def validates_all_in_cart(context):
	for x in range(len(store_values)):
		validate_items_in_cart(context, store_values[x])
